va/tlsalpn.go

```diff
@@ -163,6 +175,29 @@ func (va *ValidationAuthorityImpl) tlsDial(ctx context.Context, hostPort string,
 	return conn, nil
 }
 
+func checkExpectedSAN(cert *x509.Certificate, name identifier.ACMEIdentifier) error {
+	if len(cert.DNSNames) != 1 {
+		return errors.New("wrong number of dNSNames")
+	}
+
+	for _, ext := range cert.Extensions {
```

Comment:
Could a cert have two of the same extensions? I assume Go handles this, but have we confirmed that behavior?

Reply:
It really truly should not, and I believe that Go will complain about this, but yes this is why I don't break out of this loop after we've found the extension the first time.